nano/store/rocksdb/utility.cpp

```diff
@@ -0,0 +1,11 @@
+#include <nano/store/rocksdb/transaction_impl.hpp>
+#include <nano/store/rocksdb/utility.hpp>
+
+auto nano::store::rocksdb::tx (store::transaction const & transaction_a) -> std::variant<::rocksdb::Transaction *, ::rocksdb::ReadOptions *>
+{
+	if (dynamic_cast<nano::store::read_transaction const *> (&transaction_a) != nullptr)
+	{
+		return static_cast<::rocksdb::ReadOptions *> (transaction_a.get_handle ());
```

Comment:
Is it possible for the handle to store the pointer as `std::any` and use type safe cast `std::any_cast`?

Reply:
Yes, I think applying a similar change to transactions would be another good improvement. I'll leave that to another PR though.